### ISSUE TITLE
Question about use of `.data` in `SparseAutoencoder` class

### ISSUE THREAD

**Issue by chanind** [jbloomAus/SAELens#49]
In the SAE code, there's a number of places that read and write to `.data` instead of the tensor itself, for instance [here](https://github.com/jbloomAus/mats_sae_training/blob/main/sae_training/sparse_autoencoder.py#L67), and [here](https://github.com/jbloomAus/mats_sae_training/blob/main/sae_training/sparse_autoencoder.py#L245-L266). Why is this done, and is it necessary? It appears that reading and writing from `.data` in pytorch is deprecated and potentially dangerous: https://discuss.pytorch.org/t/the-difference-between-torch-tensor-data-and-torch-tensor/25995/2

**Comment by ArthurConmy** [comment]
I didn't know this was deprecated, thanks for sharing! 

I was using `.data` when I wanted to ensure gradients weren't tracked; it seems now we use @torch.no_grad everywhere so feel free to send a PR
